newlib/libc/machine/riscv/strcmp.S

```diff
@@ -30,29 +30,19 @@ strcmp:
 
 .size	strcmp, .-strcmp
 #else
-  li    t2, -1
 
 #if !(__riscv_misaligned_slow || __riscv_misaligned_fast)
   or    a4, a0, a1
   and   a4, a4, SZREG-1
   bnez  a4, .Lmisaligned
 #endif
 
-#if SZREG == 4
-  li a5, 0x7f7f7f7f
-#else
-  ld a5, mask
-#endif
-
   .macro check_one_word i n
     REG_L a2, \i*SZREG(a0)
 
-    and   t0, a2, a5
```

Comment:
if this was a C function, you could simply call __libc_detect_null(). Eric has already optimized this for Zbb in his patch.
I do not see anything in this function that would not work in C. Maybe this is a candidate to move from asm to C